Analysis/src/ConstraintGraphBuilder.cpp

```diff
@@ -1151,6 +1152,15 @@ ControlFlow ConstraintGraphBuilder::visit(const ScopePtr& scope, AstStatDeclareC
 
     scope->exportedTypeBindings[className] = TypeFun{{}, classTy};
 
+    if (FFlag::LuauParseDeclareClassIndexer && declaredClass->indexer)
+    {
+        // TODO: Recursion limit.
```

Comment:
Take a look at `ConstraintGraphBuilder::checkPack(const ScopePtr&, AstExpr*, const std::vector<std::optional<TypeId>>&)` for an example of how to implement this.

Reply:
This was copied from the AstTypeTable code which has the same TODO comment, but yeah I can add it in
https://github.com/Roblox/luau/blob/3ecd3a82abe6682988112cd6eb7326a67ebbc47f/Analysis/src/ConstraintGraphBuilder.cpp#L2494-L2501